specification/monitor/resource-manager/Microsoft.Insights/preview/2024-03-01-preview/metricAlert_API.json

```diff
@@ -0,0 +1,1263 @@
+            }
+          },
+          "default": {
+            "description": "BadRequest",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create or update a query based alert rule": {
+            "$ref": "./examples/createOrUpdateMetricAlertQuery.json"
+          },
+          "Create or update an alert rule for Single Resource": {
+            "$ref": "./examples/createOrUpdateMetricAlertSingleResource.json"
+          },
+          "Create or update an alert rule for Multiple Resource": {
+            "$ref": "./examples/createOrUpdateMetricAlertMultipleResource.json"
+          },
+          "Create or update an alert rule on Resource group(s)": {
+            "$ref": "./examples/createOrUpdateMetricAlertResourceGroup.json"
+          },
+          "Create or update an alert rule on Subscription ": {
+            "$ref": "./examples/createOrUpdateMetricAlertSubscription.json"
+          },
+          "Create or update a dynamic alert rule for Single Resource": {
+            "$ref": "./examples/createOrUpdateDynamicMetricAlertSingleResource.json"
+          },
+          "Create or update a dynamic alert rule for Multiple Resources": {
+            "$ref": "./examples/createOrUpdateDynamicMetricAlertMultipleResource.json"
+          },
+          "Create or update a web test alert rule": {
+            "$ref": "./examples/createOrUpdateWebTestMetricAlert.json"
+          },
+          "Create or update an alert rules with dimensions": {
+            "$ref": "./examples/createOrUpdateMetricAlertWithDimensions.json"
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "MetricAlerts"
+        ],
+        "description": "Update an metric alert definition.",
+        "operationId": "MetricAlerts_Update",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/RuleNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/MetricAlertResourcePatch"
+            },
+            "description": "The parameters of the rule to update."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/MetricAlertResource"
+            }
+          },
+          "default": {
+            "description": "BadRequest",
```

Comment:
This defines the error contract for all kinds of errors, not just for bad requests